cishouseholds/derive.py

```diff
@@ -16,6 +16,26 @@
 from cishouseholds.pyspark_utils import get_or_create_spark_session
 
 
+def derive_had_symptom_last_7days_from_digital(
+    df: DataFrame, column_name_to_assign: str, symptom_column_prefix: str, symptoms: List[str]
+):
+    """
+    Derive symptoms in v2 format from digital file
+    """
+    symptom_columns = [f"{symptom_column_prefix}_{symptom}" for symptom in symptoms]
+
+    df = count_value_occurrences_in_column_subset_row_wise(df, "NUM_NO", symptom_columns, "No")
+    df = count_value_occurrences_in_column_subset_row_wise(df, "NUM_YES", symptom_columns, "Yes")
+    df = count_value_occurrences_in_column_subset_row_wise(df, "NUM_NULL", symptom_columns, None)
+    df = df.withColumn(
+        column_name_to_assign,
+        F.when(F.col("NUM_YES") > 0, "Yes")
+        .when(F.col("NUM_NULL") == len(symptom_columns), None)
+        .when(F.col("NUM_NO") == len(symptom_columns), "No"),
```

Comment:
I've just clarified the requirement on this, and if any response(s) are "No" and the remainder are Null the outcome should be "No". So this could be simplified to use the same > 0 logic as "Yes" and then ommit the null count and condition all together (the default `.otherwise` is `None`)